junit-platform-engine/src/main/java/org/junit/platform/engine/support/store/NamespacedHierarchicalStore.java

```diff
@@ -237,26 +237,34 @@ public void close() {
 	 * closed
 	 * @since 6.0
 	 */
+	@SuppressWarnings("ReferenceEquality")
 	@API(status = MAINTAINED, since = "6.0")
 	public <K, V> Object computeIfAbsent(N namespace, K key, Function<? super K, ? extends V> defaultCreator) {
-		Preconditions.notNull(defaultCreator, "defaultCreator must not be null");
```

Comment:
this Preconditions one even does enforce (for a good reason i admin) some random (crazy) string, which is kind of good of course not to be to vague and broad. 

Of course the strack trace is nice then but will always need to be debugged.

long story short i would go with convention principle here using the good old pojo.